lib/utils/device/ssh.ts

```diff
@@ -93,36 +91,30 @@ export async function performLocalDeviceSSH(
 			);
 		}
 		if (containers.length > 1) {
-			exitWithExpectedError(stripIndent`
+			throw new ExpectedError(stripIndent`
 				Found more than one container with a service name ${opts.service}.
 				This state is not supported, please contact support.
 			`);
 		}
 
-		// Getting a command to work on all platforms is a pain,
-		// so we just define slightly different ones for windows
-		if (os.platform() !== 'win32') {
-			const shellCmd = `/bin/sh -c "if [ -e /bin/bash ]; then exec /bin/bash; else exec /bin/sh; fi"`;
-			command = `'${deviceContainerEngineBinary}' exec -ti ${
-				containers[0]!.id
-			} '${shellCmd}'`;
-		} else {
-			const shellCmd = `/bin/sh -c "if [ -e /bin/bash ]; then exec /bin/bash; else exec /bin/sh; fi"`;
-			command = `${deviceContainerEngineBinary} exec -ti ${
-				containers[0]!.id
-			} ${shellCmd}`;
-		}
+		const containerId = containers[0]!.id;
+		const shellCmd = `/bin/sh -c "if [ -e /bin/bash ]; then exec /bin/bash; else exec /bin/sh; fi"`;
+		// stdin (fd=0) is not a tty when data is piped in, for example
+		// echo 'ls -la; exit;' | balena ssh 192.168.0.20 service1
+		// See https://www.balena.io/blog/balena-monthly-roundup-january-2020/#charliestipsntricks
+		//     https://assets.balena.io/newsletter/2020-01/pipe.png
+		const ttyFlag = isatty(0) ? '-t' : '';
+		command = `${deviceContainerEngineBinary} exec -i ${ttyFlag} ${containerId} ${shellCmd}`;
```

Comment:
Using `exec -i` instead of `exec -ti` fixes a bug (error "the input device is not a TTY") when a command is piped to ssh, e.g.:
```
echo 'ls -la; exit;' | balena ssh 192.168.0.20 service1
```